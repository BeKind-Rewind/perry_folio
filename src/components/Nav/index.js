import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { capitalizeFirstLetter } from "../../utils/helpers";


function Nav(props) {
  const {
    categories = [],
    setCurrentCategory,
    currentCategory
  } = props;

  useEffect(() => {
    document.title = capitalizeFirstLetter(currentCategory.name);
  }, [currentCategory]);



  return (
    <header className="flex-row px-1">
      <h2>
        <a data-testid="link" href="/">
          Amanda Perry
        </a>
      </h2>
      <nav>
        <ul className="flex-row">
          {categories.map((category) => (
            <li
              className="mx-1"
              key={category.name}
            >
              <span
                onClick={() => {
                  setCurrentCategory(category);
                }}
              >
                <Link className={`${currentCategory.name === category.name && 'navActive'}`} to={category.name}>{capitalizeFirstLetter(category.name)}
                </Link>
              </span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

/* <span role="img" aria-label="camera"> ðŸ“¸</span> */

export default Nav;