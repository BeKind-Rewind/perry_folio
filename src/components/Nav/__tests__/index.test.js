import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);


describe('Nav component', () => {
  // BASELINE TEST
  it('renders', () => {
    render(<Nav />);
  });
  // SNAPSHOT TEST
  it('matches snapshot', () => {
    const { asFragment } = render(<Nav />);
    //value comparison

    expect(asFragment()).toMatchSnapshot();
  });

})

// describe('links are visible', () => {
//   it('inserts text into the links', () => {
//     const { getByTestId } = render(<Nav />);
//     expect(getByTestId('about')).toHaveTextContent('About me');
//   });
// })
