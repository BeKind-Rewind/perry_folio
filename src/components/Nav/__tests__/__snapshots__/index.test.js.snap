// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        href="/"
      >
        Perry
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Portfolio
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Resum√©
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
